sunflower-land/sunflower-land
# [feat] load farm from other wallets id

### Feature: Farms explorer.
View other farms searching by his wallet address.

First time developing with TypeScript, Web3 and xstate, so feel free to review/comment me anything.

QA:
- should open a modal on clicking explore farm button.
- should display loading message on search.
- should display new farm after search.
- should not have actionable buttons on exploring farm.
- should go to own farm after click "Go Back".
- should log in console if get blockchain data fails (could be a error text in the modal as improvement)

TODO: 

- ~~clean code~~
- ~~remove hardcoded wallet id~~
- ~~add input for wallet id (modal maybe)~~
- ~~add loading when fetching data~~
- ~~remove interactions in exploring farm~~
- ~~rewording labels~~
- ~~ask for css & design help :)~~ _nobody answered :(_
- ~~label to show actual farm address~~

actual state: 

https://user-images.githubusercontent.com/46355516/148410788-3ec8ce4c-4f6a-4c95-9e95-bb5e1ce19542.mp4

![image](https://user-images.githubusercontent.com/46355516/148410896-998825a9-189d-4a01-9865-84ad26eb4fed.png)
